Sources/Foundation/Process.swift

```diff
@@ -950,6 +957,27 @@ open class Process: NSObject, @unchecked Sendable {
         }
 #endif
 
+        // Unsafe fallback for systems missing posix_spawn_file_actions_addchdir[_np]
+        // This includes Glibc versions older than 2.29 such as on Amazon Linux 2
+        let previousDirectoryPath: String?
+        if useFallbackChdir {
+            let fileManager = FileManager()
+            previousDirectoryPath = fileManager.currentDirectoryPath
+            if let dir = currentDirectoryURL?.path, !fileManager.changeCurrentDirectoryPath(dir) {
+                throw _NSErrorWithErrno(errno, reading: true, url: currentDirectoryURL)
+            }
+        } else {
+            previousDirectoryPath = nil
+        }
+
+        defer {
+            if let previousDirectoryPath {
+                // Reset the previous working directory path.
+                let fileManager = FileManager()
+                _ = fileManager.changeCurrentDirectoryPath(previousDirectoryPath)
```

Comment:
NO idea why the test is still running, but I think it'll fail because `fileManager.changeCurrentDirectoryPath` requires `previousDirectoryPath` to be non-optional I believe.